File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기.java
import java.util.*;

class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; i++){
            arr[i] = sc.nextInt();
        }
        quickSort(arr, 0, arr.length - 1);
        for(int i = 0; i < n; i++){
            System.out.println(arr[i]);
        }
    }
    public static void quickSort(int[] arr, int l, int r){
        if(l > r)return;
        int left = l;
        int right = r;
        int pivot = right;
        while(left < right){
            while(arr[left] < arr[pivot])left++;
            swap(arr, left, pivot);
            pivot = left;
            while(arr[right] > arr[pivot])right--;
            swap(arr, right, pivot);
            pivot = right;
        }
        quickSort(arr, l, pivot - 1);
        quickSort(arr, pivot + 1, r);
    }
    public static void swap(int[] arr, int i, int j){
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}

File: 백준/Bronze/2750. 수 정렬하기/수 정렬하기.kt
import java.util.*

fun main(args: Array<String>)  = with(Scanner(System.`in`)) {
    val n = nextInt()
    val arr = IntArray(n)
    for(i in arr.indices) {
        arr[i] = nextInt()
    }
    f(0, n-1, arr)
    println(arr.joinToString("\n"))
}
fun f(stat:Int, end:Int, arr:IntArray){
    if(stat >= end) return
    var pivot = stat
    var l = stat+1
    var r = end
    while(l <= r){
        val temp = arr[l]
        arr[l] = arr[r]
        arr[r] = temp
        while(l<arr.size&&arr[l] <= arr[pivot]){l++}
        while(arr[r] > arr[pivot]){r--}
    }
    val temp = arr[r]
    arr[r] = arr[pivot]
    arr[pivot] = temp
    f(stat, r-1, arr)
    f(r+1, end, arr)
}